mozilla-detekt-rules/src/main/java/org/mozilla/fenix/detektrules/MozillaRunblockingCheck.kt

```diff
@@ -0,0 +1,38 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.detektrules
+
+import io.gitlab.arturbosch.detekt.api.*
+import org.jetbrains.kotlin.psi.*
+import org.jetbrains.kotlin.resolve.BindingContext
+import org.jetbrains.kotlin.resolve.calls.callUtil.getCall
+import org.jetbrains.kotlin.resolve.calls.callUtil.getCalleeExpressionIfAny
+import kotlin.math.exp
+
+private const val VIOLATION_MSG = "Please use `org.mozilla.fenix.CoroutineManager.fenixRunBlocking` instead because it allows " +
+        "us to monitor the code for performance regressions."
+
+/**
+ * A check to prevent us from working around mechanisms we implemented to prevent suppressing StrictMode.
+ *
+ * IF YOU UPDATE THIS FILE NAME, UPDATE CODE OWNERS.
+ */
+
+
+class MozillaRunblockingCheck(config: Config) : Rule(config) {
+
+    override val issue = Issue(
+        "MozillaRunblockingCheck",
+        Severity.Performance,
+        "Prevents us from working around mechanisms we implemented to prevent suppressing StrictMode",
+        Debt.TEN_MINS
```

Comment:
I changed it to 20min. I think I put it that way since I thought "people will probably not remove the `runBlocking` call but rather just call the `runBlockingCounter` "

Reply:
Oh, you're right – maybe this should be 5 min then. :P (is this detekt feature even useful?)